src/lib/pubkey/curve448/curve448.cpp

```diff
@@ -0,0 +1,125 @@
+/*
+* Curve448
+* (C) 2024 Jack Lloyd
+*     2024 Fabian Albert - Rohde & Schwarz Cybersecurity
+*
+* Botan is released under the Simplified BSD License (see license.txt)
+*/
+
+#include <botan/curve448.h>
+
+#include <botan/ber_dec.h>
+#include <botan/der_enc.h>
+#include <botan/rng.h>
+#include <botan/internal/curve448_internal.h>
+#include <botan/internal/pk_ops_impl.h>
+
+namespace Botan {
+
+namespace {
+void curve448_basepoint_from_data(std::span<uint8_t, 56> mypublic, std::span<const uint8_t, 56> secret) {
+   auto bp = x448_basepoint(decode_scalar(secret));
+   auto bp_bytes = encode_point(bp);
+   copy_mem(mypublic, bp_bytes);
+}
+
+secure_vector<uint8_t> ber_decode_sk(std::span<const uint8_t> key_bits) {
+   secure_vector<uint8_t> decoded_bits;
+   BER_Decoder(key_bits).decode(decoded_bits, ASN1_Type::OctetString).discard_remaining();
```

Comment:
Is `discard_remaining()` really what we want here? I see that it's implemented like that in 25519, but doesn't that allow 'inconsistent data' to be processed?

Reply:
I admit I did not loop deeper into this encoding/decoding stuff. I'll try to remove it.